src/QMCHamiltonians/SOECPComponent.h

```diff
@@ -0,0 +1,111 @@
+//////////////////////////////////////////////////////////////////////////////////////
+// This file is distributed under the University of Illinois/NCSA Open Source License.
+// See LICENSE file in top directory for details.
+//
+// Copyright (c) 2019 Jeongnim Kim and QMCPACK developers.
+//
+// File developed by: Raymond Clay, rclay@sandia.gov, Sandia National Laboratory
+//
+// File created by: Raymond Clay, rclay@sandia.gov, Sandia National Laboratory
+//////////////////////////////////////////////////////////////////////////////////////
+
+
+#ifndef QMCPLUSPLUS_SO_ECPOTENTIAL_COMPONENT_H
+#define QMCPLUSPLUS_SO_ECPOTENTIAL_COMPONENT_H
+#include "QMCHamiltonians/QMCHamiltonianBase.h"
+#include "QMCWaveFunctions/TrialWaveFunction.h"
+#include "Numerics/OneDimGridBase.h"
+#include "Numerics/OneDimGridFunctor.h"
+#include "Numerics/OneDimLinearSpline.h"
+#include "Numerics/OneDimCubicSpline.h"
+#include "Numerics/OhmmsBlas.h"
+
+namespace qmcplusplus
+{
+/** class SOECPComponent
+ **  brief Computes the nonlocal spin-orbit interaction $\Delta V_SO(r) |ljm_j><ljm_j|$.
+ **  details This computes the nonlocal spin-orbit interaction between a single ion species and 
+ **           a given electron.  
+ **           Currently, this class does nothing other than generate and store $\Delta V_SO(r)$
+ **           for different orbital angular momenta.  Implementation coming soon!  
+ **/
+class SOECPComponent : public QMCTraits
+{
+private:
+  typedef OneDimGridBase<RealType> GridType;
+  typedef OneDimCubicSpline<RealType> RadialPotentialType;
+
+  ///Non Local part: angular momentum, potential and grid
+  int lmax;
+  ///the number of non-local channels
+  int nchannel;
+  ///Maximum cutoff the non-local pseudopotential
+  RealType Rmax;
+  ///Angular momentum map
+  aligned_vector<int> angpp_m;
+  ///Non-Local part of the pseudo-potential
+  std::vector<RadialPotentialType*> sopp_m;
+
+public:
+  SOECPComponent(){};
+
+  ///destructor
+  ~SOECPComponent(){};
+
+  SOECPComponent* makeClone(const ParticleSet& qp){};
+
+  ///add a new Spin-Orbit component
+  void add(int l, RadialPotentialType* pp);
+  ///API for accessing the value of an SO radial potential at distance r.  For unit and other testing.
+  RealType getSplinedPotValue(int l, RealType r);
```

Comment:
If this is for unit tests only, moving it to unit tests and adding a friend function declaration are the preferred way.

Reply:
It is done.